### PR TITLE
Feat: Add NNS Index Canister to config

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Migrate ICP wallet to use the ICP index canister instead of the NNS dapp canister.

In this PR, add `NNS_INDEX_CANISTER_ID` to the frontend config.

# Changes

* Add `VITE_NNS_INDEX_CANISTER_ID ` to config.js
* Add `nns-index` entry to dfx.json

# Tests

* Update config tests with new variable

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

**Comment by lmuntaner** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #4506** <a href="https://app.graphite.dev/github/pr/dfinity/nns-dapp/4506?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dfinity/nns-dapp/4506?utm_source=stack-comment).

**Comment by lmuntaner** [comment]
@dskloetd please review

File: config.sh [hunk 1]
```diff
@@ -81,6 +81,10 @@ local_deployment_data="$(
   export CKBTC_MINTER_CANISTER_ID
   test -n "${CKBTC_MINTER_CANISTER_ID:-}" || unset CKBTC_MINTER_CANISTER_ID
 
+  : "Try to find the NNS Index canister ID"
```
Comment (dskloetd) [inline config.sh:84]: How about moving this down just below `LEDGER_CANISTER_ID`?

File: config.sh [hunk 2]
```diff
@@ -81,6 +81,10 @@ local_deployment_data="$(
...
+  NNS_INDEX_CANISTER_ID="$(dfx canister --network "$DFX_NETWORK" id nns-index 2>/dev/null || true)"
```
Comment (dskloetd) [inline config.sh:85]: Here `LEDGER_CANISTER_ID` does not have the `NNS` prefix so we don't need it for consistency.
Shall we call it just `INDEX_CANISTER_ID` here?

File: dfx.json [hunk 3]
```diff
@@ -25,6 +25,19 @@
         }
       }
     },
+    "nns-index": {
```
Comment (dskloetd) [inline dfx.json:28]: Yes, here we indeed keep `nns-`, because all the other canisters have it too.

**Review by dskloetd — APPROVED** [review]
Thanks!

File: config.sh (hunk 2 above)
Comment (lmuntaner) [reply config.sh:85]: I know it doesn't, but I believe we should change that. Just `INDEX_CANISTER_ID` seems a bit confusing to me.

File: config.sh (hunk 1 above)
Comment (lmuntaner) [reply config.sh]: done, makes sense

File: config.sh (hunk 2 above)
Comment (dskloetd) [reply config.sh]: But it should be consistent or it's also confusing.
And if we're not bound to `NNS` shouldn't it be `ICP` in order not to be confusing?

Comment (lmuntaner) [reply config.sh]: Ok, I'll keep it consistent then.

**Pull request merged**